analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -141,12 +151,12 @@ public ProgramState ResetStaticFieldConstraints(INamedTypeSymbol containingType)
 
         public override int GetHashCode() =>
             // VisitCount is not included, it's not part of Equals
-            HashCode.Combine(
-                HashCode.DictionaryContentHash(OperationValue),
-                HashCode.DictionaryContentHash(SymbolValue),
-                HashCode.DictionaryContentHash(CaptureOperation),
-                HashCode.EnumerableUnorderedContentHash(PreservedSymbols),
-                HashCode.EnumerableOrderedContentHash(Exceptions));
+            hashCode ??= HashCode.Combine(
```

Comment:
We'd need a performance comment here, explaining the overall conclusion of this PR... and why every new field should have its HashCode